web/app.js

```diff
@@ -2618,6 +2620,19 @@ function webViewerKeyUp(evt) {
   }
 }
 
+function beforeUnload(evt) {
+  // Unsaved form: are you sure?
+  if (
+    PDFViewerApplication.pdfDocument &&
+    PDFViewerApplication.pdfDocument.annotationStorage.modified
+  ) {
+    evt.preventDefault();
+    evt.returnValue = "";
```

Comment:
I'm not entirely sure what this does, but what I _think_ should happen is that a dialog is opened, similar to the document properties/password dialogs, that asks the user to confirm/cancel. The strings used should be localized by adding them to the `en-US` locale file (check how the other dialogs use localized strings).

Reply:
This appears to already have strings associated? I get `This page is asking you to confirm that you want to leave - data you have entered may not be saved.` after modifying a form and navigating away.